lib/ditcher.js

```diff
@@ -613,6 +615,63 @@ Ditcher.prototype.doImport = function (params, cb) {
   });
 };
 
+// Renames a collection while keeping the naming conventions for
+// shared dbs intact
+Ditcher.prototype.doRenameCollection = function (params, callback) {
+  if (!checkParams(params)) {
+    return callback(new Error("Invalid Params"));
+  }
+
+  var fromColl = constructCollectionName(params);
+
+  // Need to switch the `type` parameter now because `constructCollectionName` expects
+  // that one to be the collection name
+  params.type = params.toCollection;
+  var toColl = constructCollectionName(params);
+
+  this.database.renameCollection(fromColl, toColl, params, callback);
+};
+
+// Returns a native MongoDB collection instance which can be used for update/insert/find
+// operations. This function adheres to the shared db naming conventions.
+Ditcher.prototype.doGetCollectionInstance = function (params, callback) {
+  if (!checkParams(params)) {
+    return callback(new Error("Invalid Params"));
+  }
+
+  if (params.type && params.type.length > MAX_COLLECTION_NAME) {
+    return callback("Error: 'type' name too long: '" + params.type + "'. Collection name cannot be greater than: " + MAX_COLLECTION_NAME);
+  }
+
+  var coll = constructCollectionName(params);
+  return this.database.getCollectionInstance(coll, params, callback);
+};
 
+// Returns a cursor that can be used for listing the collections. This function
+// adheres to the shared-db naming conventions. Also accepts a filter object to
+// filter the returned collections.
+Ditcher.prototype.doGetCollectionsListCursor = function (filter, params) {
+  var self = this;
+
+  if (!params.__fhdb ||
+    (!params.__dbperapp && !appname_regex.test(params.__fhdb)) ||
+    (params.__dbperapp ===true && params.__fhdb !== self.database.name)) {
+    console.log(params);
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this logging statement. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS2228)